solidity_contracts/src/L1L2Messaging/L1KakarotMessaging.sol

```diff
@@ -28,7 +32,7 @@ contract L1KakarotMessaging {
     function sendMessageToL2(address to, uint248 value, bytes calldata data) external payable {
         uint256 totalLength = data.length + 4;
         uint256[] memory convertedData = new uint256[](totalLength);
-        convertedData[0] = uint256(uint160(msg.sender));
+        convertedData[0] = uint256(uint160(AddressAliasHelper.applyL1ToL2Alias(msg.sender)));
```

Comment:
other solution suggested by @enitrat would be to do the aliasing inside Kakarot

Reply:
let's keep this mechanism in pure solidity